File: Search/maven/src/main/java/Search/InvertedIndex.java
package Search;

import java.util.ArrayList;
import java.util.HashMap;

public class InvertedIndex {
    private HashMap<String, ArrayList<Integer>> dictMap;

    public InvertedIndex(HashMap<String, ArrayList<Integer>> dictMap) {
        this.dictMap = dictMap;
    }

    public void updateTheMap(String docContent, int docID) {
        String fullToken = docContent.toLowerCase();
        String[] tokens = fullToken.split(" ");
        for (String str : tokens) {
            if (!dictMap.containsKey(str))
                dictMap.put(str, new ArrayList<>());

            if (!dictMap.get(str).contains(docID))
                dictMap.get(str).add(docID);
        }
    }
}

File: Phase05/Search/Search/Model/InvertedIndex.cs
using System.Collections.Generic;
namespace Search.Model
{
    public class InvertedIndex
    {
        private Dictionary<string, HashSet<int>> dictMap;

        public InvertedIndex(Dictionary<string, HashSet<int>> dictMap)
        {
            this.dictMap = dictMap;
        }

        public void UpdateTheMap(string docContent, int docID)
        {

            var fullToken = docContent.ToLower();
            var tokensOfThisDoc = fullToken.Split(' ');

            foreach (var str in tokensOfThisDoc)
            {
                if (!dictMap.ContainsKey(str))
                    dictMap[str] = new HashSet<int>();

                if (!dictMap[str].Contains(docID))
                    dictMap[str].Add(docID);
            }
        }

    }
}